### ISSUE TITLE
Loading saved best checkpoint and training on a different dataset with early stopping

### ISSUE THREAD

**Issue by Joris-Dprtr** [unit8co/darts#1099]
### **Describe the bug**
For transfer learning I would like to use a saved model, trained on a dataset and train it on a different dataset. Problem is as follows: since the best model has a loss from a different dataset, I almost never go past the early stopping criterium because the loss is different on the new dataset. 

### **To Reproduce**
Take a dataset
```
from pytorch_lightning.callbacks import EarlyStopping

my_stopper = EarlyStopping(monitor="val_loss", patience=10, min_delta=0.00001, mode='min')
pl_trainer_kwargs = {"callbacks": [my_stopper]}
```

```
my_model = RNNModel(
    model=...
    model_name="LSTM_source",
    ...
    save_checkpoints=True,
    pl_trainer_kwargs=pl_trainer_kwargs
)
```
```
my_model.fit(...)

best_model = RNNModel.load_from_checkpoint('LSTM_source', best=True)
```
```
different dataset

best_model.fit(dataset)
```

### **Expected behavior**
1. Start from the best checkpoint
2. train on new dataset
3. Stop training after performance does not improve from best loss on the new dataset (not the old dataset, like now)

Can we somehow clear the best perfomance marker or create a different checkpoint set? 

### **System (please complete the following information):**
Python 3.9.12
Darts 0.20.0

**Comment by hrzn** [comment]
You could try to re-init the `EarlyStopping` object, build a new PyTorch Lightning trainer, and provide this object as the `trainer` parameter of `fit()`. Let us know if that works.
